src/components/Admin/Organization/OrganizationPage.tsx

```diff
@@ -88,16 +88,17 @@ const OrganizationPage: FC<Props> = ({ organization, title, mode, paths }) => {
       const profileId = organization?.profile?.id;
       const initialReferences = organization?.profile?.references || [];
       const references = editedOrganization.profile.references || [];
-      const toRemove = initialReferences.filter(x => x.id && !references.some(r => r.id && r.id === x.id));
-      const toAdd = references.filter(x => !x.id);
+      const refToRemove = initialReferences.filter(x => x.id && !references.some(r => r.id && r.id === x.id));
+      const refToAdd = references.filter(x => !x.id);
+      const refToUpdate = references.filter(x => x.id);
       const tagsetsToAdd = editedOrganization.profile.tagsets?.filter(x => !x.id) || [];
 
-      for (const ref of toRemove) {
+      for (const ref of refToRemove) {
         await deleteReference({ variables: { input: { ID: ref.id } } });
       }
```

Comment:
This code should be removed at all. Reference add/remove operations are handled by the `ProfileReferenceSegment`.